src/control/server/instance_exec.go

```diff
@@ -131,21 +131,34 @@ func (srv *IOServerInstance) finishStartup(ctx context.Context, ready *srvpb.Not
 	return nil
 }
 
-func (srv *IOServerInstance) exit(exitErr error) {
-	srv.log.Infof("instance %d exited: %s", srv.Index(),
+func (srv *IOServerInstance) exit(ctx context.Context, exitErr error) {
+	srvIdx := srv.Index()
+
+	srv.log.Infof("instance %d exited: %s", srvIdx,
 		ioserver.GetExitStatus(exitErr))
 
+	rank, err := srv.GetRank()
+	if err != nil {
+		srv.log.Debugf("instance %d: no rank assigned", srv.Index())
```

Comment:
should log the error itself? or is there only one possible cause?

Reply:
only possible cause